relayer/chain/ethereum/listener.go

```diff
@@ -117,23 +140,64 @@ func (li *Listener) pollEventsAndHeaders(
 			}
 
 			finalizedBlockNumber := gethheader.Number.Uint64() - descendantsUntilFinal
-			var events []*outbound.ContractMessage
-
+			var channelEvents []*outbound.ContractMessage
 			for _, channelContract := range li.contracts {
-				channelEvents, err := li.queryEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
+				contractEvents, err := li.queryChannelEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
 				if err != nil {
 					li.log.WithError(err).Error("Failure fetching event logs")
 				}
 
-				events = append(events, channelEvents...)
+				channelEvents = append(channelEvents, contractEvents...)
+			}
+			li.forwardChannelEvents(ctx, hcs, channelEvents)
+
+			// Query PolkadotRelayChainBridge contract's InitialVerificationSuccessful events
+			blockNumber := gethheader.Number.Uint64()
+			var events []*polkadotrelaychainbridge.ContractInitialVerificationSuccessful
+
+			contractEvents, err := li.queryLightClientEvents(ctx, blockNumber, &blockNumber)
+			if err != nil {
+				li.log.WithError(err).Error("Failure fetching event logs")
 			}
+			events = append(events, contractEvents...)
 
-			li.forwardEvents(ctx, hcs, events)
+			if len(events) > 0 {
+				li.log.Info(fmt.Sprintf("Found %d PolkadotRelayChainBridge contract events on block %d", len(events), blockNumber))
+			}
+			li.processLightClientEvents(ctx, events)
+
+			// Mark items ReadyToComplete if the current block number has passed their CompleteOnBlock number
+			items := li.db.GetItemsByStatus(store.InitialVerificationTxConfirmed)
+			if len(items) > 0 {
+				li.log.Info(fmt.Sprintf("Found %d InitialVerificationTxConfirmed items", len(items)))
+			}
+			for _, item := range items {
+				if item.CompleteOnBlock <= blockNumber {
```

Comment:
what if a block with CompleteOnBlock is received, but ends up being a short-term fork after 1 or 2 blocks and is lost?

maybe we should wait a few more blocks, like 6 blocks to ensure we use a deeper confirmed block?

Reply:
Went with @philipstanislaus's suggestion in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/bf5e0950f7bf09a85aafd3c34bd1f613052a5fc6 - was easy to add it in since the `decendantsUntilFinal` variable was already available within scope.